playbulb-candle/app.js

```diff
@@ -81,7 +88,13 @@ document.querySelector('#pulse').addEventListener('click', changeColor);
 document.querySelector('#rainbow').addEventListener('click', changeColor);
 document.querySelector('#rainbowFade').addEventListener('click', changeColor);
 
+var colorChanging = false;
+
 function changeColor() {
+  if (colorChanging) {
```

Comment:
In order to keep the sample/codelab code easier to read and write, I've decided to clear `colorChanging` in a global `unhandledrejection` event listener.